tutorials/workshop/Workshop_4_Discriminative_Model_Training.ipynb

```diff
@@ -0,0 +1,286 @@
+   "source": [
+    "from utils import upgrade_dataloaders\n",
+    "\n",
+    "datasets = []\n",
+    "datasets.append(MetalDataset(train_X_tensor, torch.LongTensor(train_marginals[:,0]))) #TODO: check \n",
+    "datasets.append(MetalDataset(dev_X_tensor, torch.LongTensor(dev_labels+1.)))\n",
+    "datasets.append(MetalDataset(test_X_tensor, torch.LongTensor(test_labels+1.)))\n",
+    "\n",
+    "dataloaders = []\n",
+    "for dataset, split in zip(datasets, [\"train\", \"valid\", \"test\"]):\n",
+    "    dataloader = DataLoader(dataset)\n",
+    "    dataloader.split = split\n",
+    "    dataloaders.append(dataloader)\n",
+    "    \n",
+    "dataloaders = upgrade_dataloaders(dataloaders)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Training LSTM Model\n",
+    "For purposes of this tutorial, we have saved a pre-trained model that was trained using probabilistic labels generated in the previous notebook. \n",
+    "\n",
+    "We define our model here and load the pretrained weights before evaluation. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "SimpleModel(name=SimpleModel)\n"
+     ]
+    }
+   ],
+   "source": [
+    "import torch.nn as nn\n",
+    "from snorkel.mtl.simple_model import SimpleModel\n",
+    "from utils import LSTMModule, EmbeddingsEncoder\n",
+    "\n",
+    "MAX_INT = train_X_tensor.max()\n",
+    "embed_size = 4\n",
+    "hidden_size = 5\n",
+    "\n",
+    "lstm_module = LSTMModule(\n",
+    "    embed_size,\n",
+    "    hidden_size,\n",
+    "    bidirectional=False,\n",
+    "    verbose=False,\n",
+    "    lstm_reduction=\"attention\",\n",
+    "    encoder_class=EmbeddingsEncoder,\n",
+    "    encoder_kwargs={\"vocab_size\": MAX_INT + 1},\n",
+    ")\n",
+    "\n",
+    "model = SimpleModel(\n",
+    "    modules=[\n",
+    "    lstm_module,\n",
+    "    nn.Linear(lstm_module.output_dim,1)],\n",
+    "    metrics = ['accuracy', 'f1', 'precision','recall'])\n",
+    "print(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Load and Score Pre-Trained Model**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Dev Set Scores\n",
+      "{'task/data_valid/valid/accuracy': 0.3624161073825503, 'task/data_valid/valid/f1': 0.5320197044334976, 'task/data_valid/valid/precision': 0.3624161073825503, 'task/data_valid/valid/recall': 1.0}\n",
+      "Test Set Scores\n",
+      "{'task/data_test/test/accuracy': 0.28289473684210525, 'task/data_test/test/f1': 0.441025641025641, 'task/data_test/test/precision': 0.28289473684210525, 'task/data_test/test/recall': 1.0}\n"
+     ]
+    }
+   ],
+   "source": [
+    "model.load('./trained_spouse_model')\n",
```

Comment:
Are we going to commit this?

Reply:
its in the data download link, not committing